router/broker_test.go

```diff
@@ -38,7 +38,7 @@ func (p *testPeer) IsLocal() bool { return true }
 
 func TestBasicSubscribe(t *testing.T) {
 	// Test subscribing to a topic.
-	broker := newBroker(logger, false, true, debug, nil)
+	broker := newBroker(logger, false, true, debug, nil, nil)
```

Comment:
Looking at this makes it clear that the signature is broken. I think we should  repackage all the arguments into an non-pointer option type that the callers have to pass. This avoids compatibility problems, avoids magic `nil` `true` and `false` problems, *and* makes it easier to extend over time.

Reply:
Oh well... Big deal! I'm not against that. But I think we should discuss that in a separate issue/PR in the origin repo including the author.